backend/src/core/utils/aws-endpoint.ts

```diff
@@ -0,0 +1,19 @@
+import convict from 'convict'
+
+interface AWSEndpointConfig { aws: { awsEndpoint: any, awsRegion: any } }
+
+/**
+ * Provide an Object containing either the endpoint to send AWS requests to,
+ * or if that is absent, the region from which to generate the endpoint. 
+ * This Object is to be injected into AWS JavaScript clients.
+ * 
+ * @param config - a convict object whose schema includes AWSEndpointConfig
+ * @return an object containing either the endpoint or region field
+ */
+const configureEndpoint = <T extends AWSEndpointConfig>(config: convict.Config<T>) => {
```

Comment:
why did you choose to pass the config in, instead of importing it?

Reply:
passing it as an argument makes it more explicit that the purpose of the function is to derive something from the config. It also grants the user more control, so that if needed, the config could be changed before calling the function. Lastly, it also helps somewhat with testing, as there's one less thing to worry about when mocking config